frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.tsx

```diff
@@ -0,0 +1,223 @@
+    column: RemappingHydratedDatasetColumn,
+    targetElement: HTMLElement,
+  ) => {
+    if (column) {
+      onShowWidget(
+        {
+          id: "column_settings",
+          props: {
+            initialKey: getColumnKey(column),
+          },
+        },
+        targetElement,
+      );
+    }
+  };
+
+  const getPartitionType = (
+    partitionName: keyof ColumnNameColumnSplitSetting,
+  ) => {
+    switch (partitionName) {
+      case "rows":
+      case "columns":
+        return "dimension";
+      default:
+        return "metric";
+    }
+  };
+
+  const handleDragEnd: OnDragEndResponder = ({ source, destination }) => {
+    if (!source || !destination) {
+      return;
+    }
+    const { droppableId: sourcePartition, index: sourceIndex } = source;
+    const { droppableId: destinationPartition, index: destinationIndex } =
+      destination;
+
+    if (
+      sourcePartition === destinationPartition &&
+      sourceIndex !== destinationIndex
+    ) {
+      onChange({
+        ...value,
+        [sourcePartition]: columnMove(
+          value[sourcePartition as keyof ColumnNameColumnSplitSetting],
+          sourceIndex,
+          destinationIndex,
+        ),
+      });
+    } else if (sourcePartition !== destinationPartition) {
+      const column =
+        value[sourcePartition as keyof ColumnNameColumnSplitSetting][
+          sourceIndex
+        ];
+
+      onChange({
+        ...value,
+        [sourcePartition]: columnRemove(
+          value[sourcePartition as keyof ColumnNameColumnSplitSetting],
+          sourceIndex,
+        ),
+        [destinationPartition]: columnAdd(
+          value[destinationPartition as keyof ColumnNameColumnSplitSetting],
+          destinationIndex,
+          column,
+        ),
+      });
+    }
+  };
+
+  const updatedValue = useMemo(
+    () =>
+      _.mapObject(value || {}, columnNames =>
+        columnNames
+          .map(columnName => columns.find(col => col.name === columnName))
+          .filter((col): col is RemappingHydratedDatasetColumn => col != null),
+      ),
+    [columns, value],
+  );
+
+  return (
+    <DragDropContext onDragEnd={handleDragEnd}>
+      {partitions.map(({ name: partitionName, title }, index) => {
+        const updatedColumns = updatedValue[partitionName] ?? [];
+        const partitionType = getPartitionType(partitionName);
+        return (
+          <Box
+            className={cx(CS.py2, { [CS.borderTop]: index > 0 })}
+            key={partitionName}
+          >
+            <Text c="text-medium">{title}</Text>
+            <Droppable droppableId={partitionName} type={partitionType}>
+              {(provided, snapshot) => (
+                <Box
+                  {...provided.droppableProps}
+                  bg={snapshot.draggingFromThisWith ? "border" : "none"}
+                  ref={provided.innerRef}
+                  mih="2.5rem"
+                  pos="relative"
+                  style={{ borderRadius: theme.radius.md }}
```

Comment:
note: you can use `var(--mantine-radius-md)` to avoid importing the theme object in Mantine 6